pydantic/_internal/_fields.py

```diff
@@ -41,6 +42,13 @@ class PydanticMetadata(Representation):
     __slots__ = ()
 
 
+@dataclasses.dataclass(**slots_true)  # TODO: make kw_only when we drop support for 3.9.
+class PydanticExtraInfo:
+    # TODO: make use of PEP 747:
+    annotation: Any
```

Comment:
`typing_extensions.TypeForm`? Or is the point that we'll need to update places across the code for this?